apps/studio/src/features/editing-experience/components/form-builder/FormBuilder.tsx

```diff
@@ -53,32 +56,39 @@ const renderers: JsonFormsRendererRegistryEntry[] = [
 ]
 
 export interface FormBuilderProps {
-  component: keyof typeof IsomerSchema.components.complex
+  component: IsomerComplexComponentProps['type']
 }
 
 export default function FormBuilder({
   component,
 }: FormBuilderProps): JSX.Element {
-  const { properties, ...rest } = IsomerSchema.components.complex[component]
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const { type, ...props } = properties
-  const schema = {
-    ...rest,
-    properties: props,
-    components: IsomerSchema.components,
-  }
-
+  const [jsonSchema, setJsonSchema] = useState<IsomerJsonSchema | null>(null)
   const [formData, setFormData] = useState({})
 
+  trpc.page.getIsomerJsonSchema.useQuery(undefined, {
```

Comment:
this kind of seems like a code smell to me tbh. this is because we're calling it and using `setState` in a callback of `onSuccess` which:
1. seems very convoluted
2. why not just use the return directly? 

should we be using `useSuspenseQuery` here instead? it returns `[data, query]` directly and allows us to act like the query returns a value instead of a promise. (see [here](https://tanstack.com/query/v4/docs/framework/react/guides/suspense))

Reply:
Moved to the EditorDrawerContext in 71a18021a67d22a779bab79620dc28904bb1c7b1. This fetch will happen once when the user first enters an editing page and I think will be cached automatically by react-query.